### PR TITLE
Gradle integration test phase

### PR DESCRIPTION
**Pull request by atomfrede**
Gradle now offers `test` and `integrationTest` like proposed using failsafe for maven here https://github.com/jhipster/generator-jhipster/pull/9174

The `check` tasks depends on all tests tasks, so `./gradlew check` executes both unit and integration tests. The results in sonar look the same as before.

I have added the `integrationTest` task as a separate pipeline step in gitlab-ci and added the integration test to the normal test step in all other ci providers for now.

screenshot_2019-02-22 gfdsg

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by DanielFran** [comment]
@atomfrede I am aligning maven with you changes for gradle.
Note: instead of using _sonar.genericcoverage.unitTestReportPaths_ it should be since SQ 6.2 _sonar.coverageReportPaths_ (see https://github.com/SonarSource/SonarJS/issues/751)

**Comment by atomfrede** [comment]
@DanielFran Thanks for the deprecation hint. Strange enough the output suggested `sonar.genericcoverage.unitTestReportPaths` as the even older property is deprecated, afterwards the same output was there. Will change it. Adn yes, looks like I forget `test-integration/scripts/21-tests-backend.sh1`. Going to update the PR.

**Comment by atomfrede** [comment]
Using `unitTestReportPaths` seems not to work for me. `Error during parsing of the generic coverage report '/home/fred/git/github/jhipster/gradle-integration-test/build/test-results/jest/TESTS-results-sonar.xml'. Look at SonarQube documentation to know the expected XML format`

**Comment by DanielFran** [comment]
@atomfrede Maybe we should create constants for all those properties?

File: generators/server/templates/gradle/sonar.gradle.ejs [hunk 1]
```diff
@@ -54,10 +54,19 @@ sonarqube {
...
         property "sonar.java.codeCoveragePlugin", "jacoco"
+        <%_ if (!skipClient) { _%>
         property "sonar.typescript.lcov.reportPaths", "${project.buildDir}/test-results/lcov.info"
-        property "sonar.junit.reportPaths", "${project.buildDir}/test-results"
+        property "sonar.testExecutionReportPaths", "${project.buildDir}/test-results/jest/TESTS-results-sonar.xml"
+        <%_ } _%>
+        property "sonar.junit.utReportFolder", "${project.buildDir}/test-results/test"
+        property "sonar.junit.itReportFolder", "${project.buildDir}/test-results/integrationTest"
+        property "sonar.junit.reportPaths", "${project.buildDir}/test-results/test,${project.buildDir}/test-results/integrationTest"
+        property "sonar.sources", "${project.projectDir}/src/main/"
```
Comment (DanielFran) [inline generators/server/templates/gradle/sonar.gradle.ejs]: Duplicated properties ;)

Comment (atomfrede) [reply generators/server/templates/gradle/sonar.gradle.ejs]: :+1:

**Comment by atomfrede** [comment]
> @atomfrede Maybe we should create constants for all those properties?

You mean for the property keys or the values or both? But yes somehow we should try to centralize that for easier maintenance after we have merged both failsafe and this

**Comment by DanielFran** [comment]
Yes, properties...

**Comment by atomfrede** [comment]
If no one has any objections I would merge this.

**Comment by pascalgrimaud** [comment]
@atomfrede : I'd prefer to merge both (maven + gradle) at the same time. But the PR for Maven has still some issue

**Comment by atomfrede** [comment]
:heavy_check_mark: No problem

**Comment by DanielFran** [comment]
As suggested by @pascalgrimaud I am merging it.

**Pull request merged**